langgraph/checkpoint/memory.py

```diff
@@ -188,6 +239,30 @@ async def alist(self, config: RunnableConfig) -> AsyncIterator[CheckpointTuple]:
             except StopIteration:
                 return
 
+    async def asearch(
+        self,
+        metadata_query: CheckpointMetadata,
+    ) -> AsyncIterator[CheckpointTuple]:
+        """Asynchronous version of search.
+
+        This method is an asynchronous wrapper around search that runs the synchronous
+        method in a separate thread using asyncio.
+        """
+        loop = asyncio.get_running_loop()
+        iter = await loop.run_in_executor(None, self.search, metadata_query)
+
+        def next_item(iter: Iterator[CheckpointTuple]) -> CheckpointTuple:
```

Comment:
I originally tried to copy the implementation from `alist()`, but it was not working as expected since `StopIteration` is raised in the subprocess, not the main thread.

I'm not sure if `alist()` actually works as expected.

Reply:
signature seems wrong? also this function seems equivalent to `next(iter, None)`

Reply:
I'll adopt the implementation from this PR: https://github.com/langchain-ai/langgraph/pull/447